.github/workflows/test-e2e.yml

```diff
@@ -0,0 +1,95 @@
+name: test-e2e
+on:
+  pull_request:
+    paths-ignore:
+      - 'README.md'
+      - 'CHANGELOG.md'
+      - '.changelog/**'
+      - '.tours/**'
+      - 'contributing/**'
+      - 'demo/**'
+      - 'dev/**'
+      - 'e2e/terraform/**'
+      - 'e2e/ui/**'
+      - 'integrations/**'
+      - 'pkg/**'
+      - 'scripts/**'
+      - 'terraform/**'
+      - 'ui/**'
+      - 'website/**'
+  push:
+    branches:
+      - main
+      - release/**
+      - convert-hashicorp-nomad-to-actions-20230505-212504
+    paths-ignore:
+      - 'README.md'
+      - 'CHANGELOG.md'
+      - '.changelog/**'
+      - '.tours/**'
+      - 'contributing/**'
+      - 'demo/**'
+      - 'dev/**'
+      - 'e2e/terraform/**'
+      - 'e2e/ui/**'
+      - 'integrations/**'
+      - 'pkg/**'
+      - 'scripts/**'
+      - 'terraform/**'
+      - 'ui/**'
+      - 'website/**'
+
+jobs:
+  build-binaries:
```

Comment:
You know... I have no idea. I'm gonna wait to see if Emily knows. But upon reading this workflow, it accomplishes nothing

Reply:
Ah this is the part where to build binaries for the `nomad-e2e` repo to read 😄

Reply:
I suppose we could just grab built artifacts off of the `build.yml` build as well - although I'm not entirely sure that the build processes are the same, so it could introduce unexpected behaviour.
If we do keep this, I think it's probably worth adding a note as to why these builds appear to be unused.

Reply:
Myself, I like the idea of us pulling build.yml's artifacts since then we'd be e2e testing the binaries we'd ship